config/config.go

```diff
@@ -698,6 +698,11 @@ func (c *Config) addParser(parentname string, table *ast.Table) (*models.Running
 			dataformat = "influx"
 		}
 	}
+	var influxParserType string
+	c.getFieldString(table, "influx_parser_type", &influxParserType)
+	if influxParserType == "upstream" {
+		dataformat = "influx_upstream"
+	}
```

Comment:
How about dealing with this here completely like
```suggestion
	var influxParserType string
	c.getFieldString(table, "influx_parser_type", &influxParserType)
	if dataformat == "influx" && influxParserType == "upstream" {
		dataformat = "influx_upstream"
	}
```
This should automatically solve the issue without the need to pass the parser type down the line...

Reply:
ok! changes pushed!